lib/promenade/karafka/statistics_subscriber.rb

```diff
@@ -0,0 +1,81 @@
+require "promenade/karafka/subscriber"
+require "active_support/core_ext/hash"
+
+module Promenade
+  module Karafka
+    class StatisticsSubscriber < Subscriber
+      attach_to "statistics.karafka"
+
+      Promenade.histogram :kafka_connection_latency do
+        doc "Request latency (rtt)"
+        buckets :network
+      end
+
+      Promenade.counter :kafka_connection_calls do
+        doc "Count of calls made to Kafka broker"
+      end
+
+      Promenade.gauge :kafka_consumer_ofset_lag do
+        doc "Lag between message create and consume time"
+      end
+
+      def emitted(event)
+        group = event.payload[:consumer_group_id]
+        statistics = event.payload[:statistics].with_indifferent_access
+
+        report_topic_metrics(statistics, group)
+        report_connection_metrics(statistics)
+      end
+
+      private
+
+        def report_topic_metrics(statistics, group)
+          statistics[:topics].map do |topic_name, topic_values|
+            labels = {
+              client: statistics[:client_id],
+              topic: topic_name,
+              group: group,
+            }
+            report_partition_metrics(topic_values, labels)
+          end
+        end
+
+        def report_partition_metrics(topic_values, labels)
+          topic_values[:partitions].map do |partition_name, partition_values|
+            next if partition_name == "-1"
+            next if partition_values[:consumer_lag_stored] == -1
+
+            labels = labels.merge(
+              partition: partition_name,
+            )
+
+            offset_lag = partition_values[:consumer_lag_stored]
+
+            $stdout.puts "[Statistics][karafka Topics] #{labels} - Offset Lag: #{offset_lag}"
+
+            Promenade.metric(:kafka_consumer_ofset_lag).set(labels, offset_lag)
+          end
+        end
+
+        def report_connection_metrics(statistics)
```

Comment:
Similar to `report_connection_metrics` we can reduce the knowledge of this method, looks like it only needs to handle client_id and brokers values

```suggestion
        def report_connection_metrics(client_id, broker_statistics)
```

```ruby
report_connection_metrics(statistics[:client_id], statistics[:brokers])
```